### PR TITLE
[DOCS] Adds anomaly detection resources and advanced concepts sections

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/stack-docs/pull/1726

Requires https://github.com/elastic/stack-docs/pull/1731

### Screenshots

image

### CODE REVIEW

File: docs/en/stack/ml/anomaly-detection/index.asciidoc [hunk 1]
```diff
@@ -28,68 +26,72 @@ include::stopping-ml.asciidoc[leveloffset=+2]
...
+
 include::anomaly-detection-scale.asciidoc[leveloffset=+2]
 
 include::ml-api-quickref.asciidoc[leveloffset=+1]
 
-include::ootb-ml-jobs.asciidoc[leveloffset=+1]
+include::anomaly-examples.asciidoc[leveloffset=+1]
 
-include::ootb-ml-jobs-apache.asciidoc[leveloffset=+2]
+include::{es-repo-dir}/ml/anomaly-detection/ml-configuring-alerts.asciidoc[leveloffset=+2]
```
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/index.asciidoc:37]: Would it be worth it to still keep these files in the ES repo? Is it feasible to move them in the stack-docs repo in a separate PR after the reorg is done? So all of the AD-related conceptual docs files would be in the same place.

**Review by szabosteve — APPROVED** [review]
I have only one question, otherwise LGTM after the appendices are fixed! Thanks!

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by lcawl** [comment]
@elasticmachine update branch

File: docs/en/stack/ml/anomaly-detection/index.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/en/stack/ml/anomaly-detection/index.asciidoc:37]: It's definitely something to consider, but I'll defer that decision for now.

**Pull request merged**